File: java/src/main/java/com/github/propi/rdfrules/java/data/QuadsOps.java
package com.github.propi.rdfrules.java.data;

/**
 * Created by Vaclav Zeman on 14. 1. 2020.
 */
public interface QuadsOps<SColl, JColl> {

    com.github.propi.rdfrules.data.ops.QuadsOps<SColl> asScala();

    JColl asJava(SColl scala);

}

File: core/src/main/scala/com/github/propi/rdfrules/data/ops/QuadsOps.scala
package com.github.propi.rdfrules.data.ops

import com.github.propi.rdfrules.data.Quad
import com.github.propi.rdfrules.data.Quad.QuadTraversableView

/**
  * Created by Vaclav Zeman on 14. 1. 2020.
  */
trait QuadsOps[Coll] {

  def quads: QuadTraversableView

  /**
    * For all quads apply string interning
    *
    * @return
    */
  def intern: Coll = transformQuads(quads.map(_.intern))

  protected def transformQuads(col: Traversable[Quad]): Coll

}